helm/grafana-agent/templates/cilium-networkpolicy.yaml

```diff
@@ -7,30 +7,42 @@ metadata:
   labels:
     {{- include "grafana-agent.labels" . | nindent 4 }}
 spec:
+  egress:
+  - toEntities:
+    - kube-apiserver
+    - world
+  - toEndpoints:
+    - matchLabels:
+        io.kubernetes.pod.namespace: kube-system
+        k8s-app: coredns
+    - matchLabels:
+        io.kubernetes.pod.namespace: kube-system
+        k8s-app: k8s-dns-node-cache
+    toPorts:
+    - ports:
+      - port: "1053"
+        protocol: UDP
+      - port: "1053"
+        protocol: TCP
+      - port: "53"
+        protocol: UDP
+      - port: "53"
+        protocol: TCP
+  - toEndpoints:
```

Comment:
First thing, I'm not sure how this is restricted. We just remove access to the cluster but we can access outside.

But also the grafana agent rule is a custom app so I would probably deploy the CNP along with it, probably as an extra object because that means agent in workload clusters would be allowed to discussed with a non existent mimir?

Reply:
For what I understood, the restriction is on communication into the cluster as we removed 
```
- toEntities:
  - cluster
```

Reply:
Ok that make some sense but maybe we should use a custom CNP then because I think this specific one does not need World I think

Reply:
It's more difficult that it seems...
Pawel and I spent a lot of times on restricted that CNP and finally he had to work on it deeply to find the correct CNP.
So I wouldn't change it.

Instead, I thought of making the CNP customizable by the values, hence we could set specific egress/ingress in our grafana-agent-rules. 
WDYT ?

Reply:
Sure something like an extraEgress or smth is good :)

Reply:
In our case, we need to completely override the default egress.